source/opt/loop_dependence.cpp

```diff
@@ -1111,28 +1134,6 @@ PartitionedSubscripts LoopDependenceAnalysis::PartitionSubscripts(
   return partitions;
 }
 
-bool IsConstant(const SENode* node) {
-  return node->AsSEConstantNode() != nullptr;
-}
-
-// Compare 2 fractions while first normalizing them, e.g. 2/4 and 4/8 will both
-// be simplified to 1/2 and then determined to be equal.
-bool NormalizeAndCompareFractions(int64_t numerator_0, int64_t denominator_0,
-                                  int64_t numerator_1, int64_t denominator_1) {
-  auto gcd_0 =
-      GreatestCommonDivisor(std::abs(numerator_0), std::abs(denominator_0));
-  auto gcd_1 =
-      GreatestCommonDivisor(std::abs(numerator_1), std::abs(denominator_1));
-
-  auto normalized_numerator_0 = numerator_0 / gcd_0;
-  auto normalized_denominator_0 = denominator_0 / gcd_0;
-  auto normalized_numerator_1 = numerator_1 / gcd_1;
-  auto normalized_denominator_1 = denominator_1 / gcd_1;
-
-  return normalized_numerator_0 == normalized_numerator_1 &&
-         normalized_denominator_0 == normalized_denominator_1;
-}
-
 std::shared_ptr<Constraint> LoopDependenceAnalysis::IntersectConstraints(
```

Comment:
I forgot to comment on the use of shared_ptr<> before.  Is it absolutely necessary to use it?  The coding guideline strongly prefers unique_ptr with ownership transfer.  It's easier to understand than shared_ptr.